fcrepo-clients/src/test/java/edu/unc/lib/dl/event/RepositoryPremisLoggerIT.java

```diff
@@ -166,4 +181,221 @@ public void getEventsModelForObjectWithoutLog() throws Exception {
 
         assertTrue(eventsModel.isEmpty());
     }
+
+    @Test
+    public void makeMultipleChangesSimultaneously() throws InterruptedException {
+        parentObject = repoObjFactory.createDepositRecord(null);
+        initPremisLogger(parentObject);
+
+        // make sure that there are no events in premis log before the writes
+        PremisLogger retrieveLogger = new RepositoryPremisLogger(parentObject, mockSession,
+                pidMinter, repoObjLoader, repoObjFactory);
+        Model initialLogModel = retrieveLogger.getEventsModel();
+        assertFalse("New premis already contains events", initialLogModel.listObjects().hasNext());
+
+        // add original event
+        Resource originalEventResc = logger.buildEvent(Premis.note)
+                .addEventDetail("first premis event")
+                .write();
+
+        // add new events
+        List<Thread> threads = new ArrayList<>();
+        List<Resource> events = new ArrayList<>();
+        List<String> eventUris = new ArrayList<>();
+
+        Resource event1Resc = logger.buildEvent(Premis.VirusCheck)
```

Comment:
event1 and 2 here seem a little superfluous, seems like they are testing the same thing as the events from the loop. Maybe we should just have originalEvent and then all the events from the loop? It'd make the test a bit shorter and easier to understand what's being tested.